libcontainer/system/proc.go

```diff
@@ -25,3 +30,30 @@ func GetProcessStartTime(pid int) (string, error) {
 	// (divide by sysconf(_SC_CLK_TCK)).
 	return parts[22-1], nil // starts at 1
 }
+
+// GetProcessState reads /proc/<pid>/stat to determine the "state" of a process
+// as provided by the kernel. The format of the state is defined by proc(5) and
+// is not parsed by us, though it is usually a single character mnemonic.
+func GetProcessState(pid int) (string, error) {
+	data, err := ioutil.ReadFile(filepath.Join("/proc", strconv.Itoa(pid), "stat"))
+	if err != nil {
+		return "", err
+	}
+
+	parts := strings.Split(string(data), " ")
+	// the state field is located in pos 3
+	return parts[3-1], nil // starts at 1
+}
+
+// GetProcessParent reads reads /proc/<pid>/stat to determine the parent of a
+// process.
+func GetProcessParent(pid int) (int, error) {
+	data, err := ioutil.ReadFile(filepath.Join("/proc", strconv.Itoa(pid), "stat"))
+	if err != nil {
+		return -1, err
+	}
+
+	parts := strings.Split(string(data), " ")
+	// the state field is located in pos 4
```

Comment:
Thanks, but note that this PR is in quite a lot of flux (it doesn't really work at the moment and I'm trying to figure out why the hell processes are so hard on Linux).